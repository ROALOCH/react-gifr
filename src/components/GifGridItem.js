import React from 'react';
import PropTypes from 'prop-types';

const GifGridItem = ({ title, url }) => {
	return (
		<div className='card card-custom animate__animated animate__fadeIn'>
			<img src={url} alt={title} className='card-img-top' />
			<div className='card-body'>
				<p className='card-text'>{title}</p>
			</div>
		</div>
	);
};

GifGridItem.propTypes = {
	title: PropTypes.string.isRequired,
	url: PropTypes.string.isRequired,
};

export default GifGridItem;
