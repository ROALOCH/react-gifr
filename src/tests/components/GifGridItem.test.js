import React from 'react';
import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem/>', () => {
	const title = 'Imagen';
	const url = 'http://localhost:3000/img.jpeg';
	const wrapper = shallow(<GifGridItem title={title} url={url} />);

	test('Debe mostrar el componente correctamente', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('Debe de tener un parrafo con el title', () => {
		const p = wrapper.find('p');
		expect(p.text().trim()).toBe(title);
	});

	test('Debe de tener la imagen igual al url y alt de los props', () => {
		const img = wrapper.find('img');
		const { src, alt } = img.props();

		expect(src).toBe(url);
		expect(alt).toBe(title);
	});

	test('Debe tener la animaciÃ³n animate__fadeIn', () => {
		const div = wrapper.find('div').first();
		const { className } = div.props();

		expect(className.includes('animate__fadeIn')).toBe(true);
	});
});
