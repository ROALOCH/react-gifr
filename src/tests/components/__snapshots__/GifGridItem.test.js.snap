// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem/> Debe mostrar el componente correctamente 1`] = `
<div
  className="card card-custom animate__animated animate__fadeIn"
>
  <img
    alt="Imagen"
    className="card-img-top"
    src="http://localhost:3000/img.jpeg"
  />
  <div
    className="card-body"
  >
    <p
      className="card-text"
    >
      Imagen
    </p>
  </div>
</div>
`;
